plugins/eslint-plugin-liferay-portal/lib/rules/no-explicit-extend.js

```diff
@@ -145,25 +95,66 @@ module.exports = {
 			return {};
 		}
 
+		const pendingDeletions = new Map();
+
 		return {
+			'ArrayExpression:exit': function(node) {
+				if (pendingDeletions.has(node)) {
+					// Special case: when removing all array items, remove
+					// entire property instead.
+					if (
+						pendingDeletions.get(node).size === node.elements.length
+					) {
+						pendingDeletions.set(node.parent, node.parent);
+					} else {
+						context.report({
+							fix: fixer =>
+								fix(pendingDeletions.get(node), context, fixer),
+							messageId,
+							node,
+						});
+					}
+				}
+			},
```

Comment:
I initially just tried the stupid/simple thing and tried to call `fix` once for each `Literal` that I wanted to remove from the array; but it seems that ESLint didn't want me to mutate the same text range (ie. the range of the array) multiple times, so I had to do this big refactor to make it do a single mutation to the array at the end, which is why I am using `ArrayExpression:exit` here (and below, `Property:exit`).